src/net_processing.cpp

```diff
@@ -1308,9 +1309,37 @@ PeerManagerImpl::PeerManagerImpl(const CChainParams& chainparams, CConnman& conn
 }
 
 /**
- * Evict orphan txn pool entries based on a newly connected
- * block, remember the recently confirmed transactions, and delete tracked
- * announcements for them. Also save the time of the last tip update.
+ * If a transaction was removed from the mempool for a reason that entails it
+ * most likely will not re-enter, let the rebroadcast handler know.
+ */
+void PeerManagerImpl::TransactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRemovalReason reason, uint64_t mempool_sequence)
+{
+    if (!m_txrebroadcast) return;
+
+    switch(reason) {
+        case MemPoolRemovalReason::BLOCK:
+            // Although transactions removed for this reason will not be
+            // returned by this callback, include it here so the compiler
+            // can warn about missing cases in this switch statement.
+            // These transactions are handled by BlockConnected.
+            break;
+        case MemPoolRemovalReason::EXPIRY:
```

Comment:
In 4165cb0606e4be476651899b80525f63f6e4465a [rebroadcast] Stop tracking rebroadcast attempts for certain transactions

Would it be possible to write in a comment here "we want to avoid a situation where two peers that are unaware of new policy/consensus rules endlessly rebroadcast a transaction considered invalid by the majority of the network. Thus, the rebroadcast attempt tracker should remember transactions even after they have expired from the mempool." or something along those lines? I just remember it being confusing what "genuinely expire from majority of mempools" meant and why we don't remove from the tracker when a tx expires from mempool.

Reply:
done, thanks! I've been having a hard time explaining, added a comment along these lines.